React-Redux Bookstore - Second Milestone

Rhelli opened the pull request:
# 📚 React-Redux BookStore - Second Milestone
Hi there,

Following the finish of the first milestone, we have completed the following steps for the second milestone:
 - Prepared the file structure as specified by the Microverse guidelines
 - Implemented the `App` component which contains the `BooksList` and `BooksForm` components
 - Implemented the `BooksList` component which is connected to the Redux store and displays the initially defined books.
 -  Implemented the `Book` component which builds individual `Books` and returns them as single row objects
 - Implemented the `BookForm` component which includes all the required fields to enter a title and choose a category for a new book as well as submit it to the redux store.
 - Defined all the Redux actions and reducers
 - Initialised the redux store with an initial state.

Let us know if there are any issues with this over slack.

All the best,

Alexis and Rory

Psiale committed:
Finished first and second point of the second milestone, also added a book class

Rhelli committed:
Added bookTypes, updated actions and reducer

Rhelli committed:
Added bookTypes, updated actions and reducer:

 - Added new folder bookTypes to contain action variables
 - Created bookActions to include the create and remove book actions
 - began creating the books reducer

Rhelli committed:
Amended missing payload from bookActions and fixed the book reducer to include the action payload and reference them.

Rhelli committed:
Defined and created the Book component. Added stateful functional component and added prop checks and table formatted return

Rhelli committed:
Modified the book component, updated the bookList to try and implement the rendering of individual books in the list, added mapStateToProps.

Psiale committed:
finished the second Milestone

Rhelli committed:
Resolved eslint errors, added proptypes to the book list and fixed missing book id namespacing to resolve missing book ids

Psiale committed:
Finished point five of the second milestone, created a BooksForm with a table and a group of categories

wathigo commented on src/containers/App.js:
```diff
@@ -0,0 +1,14 @@
+function App() {
+  return (
+    <div className="App">
+      <BooksList Books={Book} />
```
We don't need to pass the Book component as the value of Books prop since we are already accessing books from the redux store in the BookList component.

wathigo commented on src/containers/App.js:
```diff
@@ -0,0 +1,14 @@
+import BooksList from './BooksList';
+import Book from '../components/Book';
+
+function App() {
```
The App component should be placed in `src/components` since it is a representational component (has no access to the store or the app state).
Let's make use of es6 arrow syntax.

wathigo commented on src/components/BooksForm.js:
```diff
@@ -0,0 +1,34 @@
+      <label htmlFor="bookCategories">
+        Categories
+        <select name="bookCategories" id="books">
+          <option value={categories[0]}>{categories[0]}</option>
+          <option value={categories[1]}>{categories[1]}</option>
+          <option value={categories[2]}>{categories[2]}</option>
+          <option value={categories[3]}>{categories[3]}</option>
+          <option value={categories[4]}>{categories[4]}</option>
+          <option value={categories[5]}>{categories[5]}</option>
+          <option value={categories[6]}>{categories[6]}</option>
```
Let's avoid code repetition by replacing this with:
```js
{
     categories.map(category => <option key={category}>{category}</option>)
}
```

wathigo reviewed (changes requested):
## Changes Required :o: 
Great job on this milestone.
Let's make a few changes before we move on. See the comments :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

Rhelli committed:
Modified and made suggested fixes and changes by TSE:

 - Remove passing Book as prop as its already passed in Redux
 - Moved App to components folder
 - Modified the categories jsx to use map to avoid reptition

Rhelli commented:
# 📚 React-Redux Bookstore - Milestone 2
Hi there,

As proposed by @wathigo above, we have made the following changes to our project:

 - Removed the passing the `Book` component as it is already passed at the highest level via the redux store
 - Moved `App.js` to the `components` folder from the `containers` folder
 - Modified the `App` function to be an ES6 arrow function.
 - Modified `BookForm.js` to use `map` to reduce the length of our JSX.

Thanks in advance,

Alexis and Rory

Rhelli committed:
Amended the App.js component to be an ES6 arrow function

Torres-ssf reviewed (approved):
### Milestone approved :+1: 
Hi team :hand: 
Great job implementing the requested changes

- [x] Bookform component will be connected with Redux on the next milestone, therefore, is considered a Container component. We need to move it to the `src/containers` directory. 

Happy coding :computer: 
**Sergio Torres**
